spec/acceptance/cli_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper_acceptance'
+
+describe 'installation of cli only' do
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      include pulpcore::cli
+      PUPPET
+    end
+  end
+
+  describe command("pulp config validate --location /etc/pulp/cli.toml") do
+    its(:exit_status) { is_expected.to eq 0 }
+    its(:stdout) { is_expected.to match(/valid pulp-cli config/) }
+  end
+end
+
+describe 'installation of server with cli' do
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      include pulpcore
+      class { 'pulpcore::cli':
+        pulpcore_url => "https://#{host_inventory['fqdn']}/",
+        cert         => "/etc/pulpcore-certs/client-cert.pem",
+        key          => "/etc/pulpcore-certs/client-key.pem",
+      }
+      PUPPET
+    end
+  end
+
+  include_examples 'the default pulpcore application'
+
+  describe command("pulp config validate --location /etc/pulp/cli.toml") do
+    its(:exit_status) { is_expected.to eq 0 }
+    its(:stdout) { is_expected.to match(/valid pulp-cli config/) }
+  end
+
+  describe command("REQUESTS_CA_BUNDLE=/etc/pulpcore-certs/ca-cert.pem pulp status") do
+    its(:exit_status) { is_expected.to eq 0 }
+    its(:stdout) { is_expected.to match(/versions/) }
+    # currently this contains a warning:
+    # SubjectAltNameWarning: Certificate for centos8-64.example.com has no `subjectAltName`, falling back to check for a `commonName` for now. This feature is being removed by major browsers and deprecated by RFC 2818.
+    # its(:stderr) { is_expected.to eq '' }
```

Comment:
I couldn't find a good way to fiddle this code to make it work.
https://github.com/theforeman/puppet-pulpcore/blob/3dd9f3796a52b6599279ca409078755aa65780f8/spec/setup_acceptance_node.pp#L53-L58

There are a few hints here on SO: https://security.stackexchange.com/questions/74345/provide-subjectaltname-to-openssl-directly-on-the-command-line

Reply:
In the past I also struggled with this. In https://github.com/ekohl/ownca/commit/2034411c5819ac62a41bcc27ffe82f1e81ac9573 I reverted to the config because `-addext` isn't an add: it replaces all extensions from the config. However, there is no config here so I'm guessing it should work.

Reply:
https://github.com/theforeman/puppet-pulpcore/pull/253